Brokerages/Bitfinex/BitfinexSubscriptionManager.cs

```diff
@@ -319,26 +319,45 @@ private void OnMessage(object sender, WebSocketMessage e)
             {
                 var token = JToken.Parse(e.Message);
 
+                webSocket.ConnectionHandler.KeepAlive(DateTime.UtcNow);
+
                 if (token is JArray)
                 {
                     var channel = token[0].ToObject<int>();
-                    // heartbeat
-                    if (token[1].Type == JTokenType.String && token[1].Value<string>() == "hb")
+
+                    if (token[1].Type == JTokenType.String)
                     {
-                        webSocket.ConnectionHandler.KeepAlive(DateTime.UtcNow);
-                        return;
+                        var type = token[1].Value<string>();
+
+                        switch (type)
+                        {
+                            // heartbeat
+                            case "hb":
+                                return;
+
+                            // trade
+                            case "tu":
+                                OnUpdate(webSocket, channel.ToStringInvariant(), token[2].ToObject<string[]>());
+                                break;
+
+                            // ignore
+                            case "te":
```

Comment:
Wondering what's `te`? Maybe we should add why we ignore it in the comment

Reply:
The `te` message is a trade execution, similar to `tu` (trade execution update):
both messages are sent for each order fill but only the `tu` message contains fields for the fees.
I'll add a comment to clarify